Core.DomainModel/Organization/StsOrganizationConnection.cs

```diff
@@ -9,30 +12,82 @@ namespace Core.DomainModel.Organization
     /// </summary>
     public class StsOrganizationConnection : Entity, IOwnedByOrganization
     {
+        public StsOrganizationConnection()
+        {
+            StsOrganizationChangeLogs = new List<StsOrganizationChangeLog>();
+        }
+
         public int OrganizationId { get; set; }
         public virtual Organization Organization { get; set; }
         public bool Connected { get; set; }
         /// <summary>
         /// Determines the optional synchronization depth used during synchronization from STS Organisation
         /// </summary>
         public int? SynchronizationDepth { get; set; }
+        public virtual ICollection<StsOrganizationChangeLog> StsOrganizationChangeLogs { get; set; }
+
 
         public bool SubscribeToUpdates { get; set; }
-        //TODO https://os2web.atlassian.net/browse/KITOSUDV-3317 adds the change logs here
+        
         public DisconnectOrganizationFromOriginResult Disconnect()
         {
             var organizationUnits = Organization.OrgUnits.Where(x => x.Origin == OrganizationUnitOrigin.STS_Organisation).ToList();
             organizationUnits.ForEach(unit => unit.ConvertToNativeKitosUnit());
+            var removedLogs = RemoveAllLogs();
 
             Connected = false;
             SubscribeToUpdates = false;
             SynchronizationDepth = null;
-            return new DisconnectOrganizationFromOriginResult(organizationUnits);
+            return new DisconnectOrganizationFromOriginResult(organizationUnits, removedLogs);
         }
 
         public IExternalOrganizationalHierarchyUpdateStrategy GetUpdateStrategy()
         {
             return new StsOrganizationalHierarchyUpdateStrategy(Organization);
         }
+
+        public StsOrganizationConnectionAddNewLogsResult AddNewLogs(IEnumerable<StsOrganizationChangeLog> newLogs)
```

Comment:
why is it a list in the param????


that just intoduces more issues that you should, but don't take into account such as
- Guarding for duplicates within the list

Also, you are missin to check for addition of the same log twice.. You can use "contains" for that...
You should also throw ArgumentNullException if the arg is null
Change from list arg ti single object arg